src/umpire/strategy/FixedPool.cpp

```diff
@@ -105,13 +105,14 @@ FixedPool::allocInPool(Pool& p) noexcept
     }
   }
 
-  UMPIRE_ASSERT("FixedPool: Logic error in allocate");
+  UMPIRE_ASSERT("FixedPool::allocInPool(): num_avail > 0, but no available slots" && 0);
   return nullptr;
 }
 
 void*
 FixedPool::allocate(size_t bytes)
 {
+  UMPIRE_ASSERT(!bytes || bytes == m_obj_bytes);
```

Comment:
Could you add a comment stating what this is asserting?  It would be helpful to understand the semantics for `bytes`.  What does it mean when `bytes == 0` and what does it mean when `bytes == m_obj_bytes`?